src/Site.js

```diff
@@ -666,32 +669,79 @@ function getPluginPath(rootPath, plugin) {
   }
 
   // Check in src folder
-  const defaultPath = path.join(__dirname, BUILT_IN_PLUGIN_FOLDER_NAME, `${plugin}.js`);
+  const srcPath = path.join(__dirname, BUILT_IN_PLUGIN_FOLDER_NAME, `${plugin}.js`);
+  if (fs.existsSync(srcPath)) {
+    return srcPath;
+  }
+
+  // Check in default folder
+  const defaultPath = path.join(__dirname, BUILT_IN_PLUGIN_DEFAULT_FOLDER_NAME, `${plugin}.js`);
   if (fs.existsSync(defaultPath)) {
     return defaultPath;
   }
 
   return '';
 }
 
+/**
+ * Finds plugins in the site's default plugin folder
+ */
+function findDefaultPlugins() {
+  const globPath = path.join(__dirname, BUILT_IN_PLUGIN_DEFAULT_FOLDER_NAME);
+  if (!fs.existsSync(globPath)) {
+    return [];
+  }
+  return walkSync(globPath, {
+    directories: false,
+    globs: ['*.js'],
+  })
+    .map(file => path.parse(file).name);
+}
+
+/**
+ * Loads a plugin
+ * @param plugin name of the plugin
+ * @param isDefault whether the plugin is a default plugin
+ */
+Site.prototype.loadPlugin = function (plugin, isDefault) {
+  try {
+    // Check if already loaded
+    if (this.plugins[plugin]) {
+      return;
+    }
+
+    const pluginPath = getPluginPath(this.rootPath, plugin);
+    if (isDefault && !pluginPath.startsWith(path.join(__dirname, BUILT_IN_PLUGIN_DEFAULT_FOLDER_NAME))) {
+      logger.warn(`Default plugin ${plugin} will be overridden`);
+    }
+
+    // eslint-disable-next-line global-require, import/no-dynamic-require
+    this.plugins[plugin] = require(pluginPath || plugin);
+  } catch (e) {
+    logger.warn(`Unable to load plugin ${plugin}, skipping`);
+  }
+};
+
 /**
  * Load all plugins of the site
  */
 Site.prototype.collectPlugins = function () {
   if (!this.siteConfig.plugins) {
-    return;
+    this.siteConfig.plugins = [];
   }
+
   module.paths.push(path.join(this.rootPath, 'node_modules'));
-  this.siteConfig.plugins.forEach((plugin) => {
-    try {
-      const pluginPath = getPluginPath(this.rootPath, plugin);
 
-      // eslint-disable-next-line global-require, import/no-dynamic-require
-      this.plugins[plugin] = require(pluginPath || plugin);
-    } catch (e) {
-      logger.warn(`Unable to load plugin ${plugin}, skipping`);
-    }
-  });
+  const defaultPlugins = findDefaultPlugins();
+
+  this.siteConfig.plugins
+    .filter(plugin => !_.includes(defaultPlugins, plugin))
+    .forEach(plugin => this.loadPlugin(plugin, false));
+  defaultPlugins
+    .filter(plugin => !this.siteConfig.pluginsContext
```

Comment:
Just wondering - would using lodash's `get` method make this more readable? 🙂 

Possibly something like this:
```js
_.get(this.siteConfig, ["pluginsContext", plugin, "off"], false)
```

Reply:
TIL, thanks